install/install.go

```diff
@@ -1445,27 +1447,71 @@ func (k *K8sInstaller) restartUnmanagedPods(ctx context.Context) error {
 		}
 	}
 
-	for _, pod := range pods.Items {
-		if !pod.Spec.HostNetwork {
-			if _, ok := cepMap[pod.Namespace+"/"+pod.Name]; ok {
-				continue
-			}
+	tail := func(pods []corev1.Pod) []corev1.Pod {
+		if len(pods) <= 1 {
+			pods = nil
+		} else {
+			pods = pods[1:]
+		}
+		return pods
+	}
 
-			if !printed {
-				k.Log("♻️  Restarting unmanaged pods...")
-				printed = true
-			}
-			err := k.client.DeletePod(ctx, pod.Namespace, pod.Name, metav1.DeleteOptions{})
-			if err != nil {
-				k.Log("⚠️  Unable to restart pod %s/%s: %s", pod.Namespace, pod.Name, err)
-			} else {
+	// group pods by namespace to make evictions faster
+	nsToPods := make(map[string][]corev1.Pod)
+	for _, p := range pods.Items {
+		nsToPods[p.Namespace] = append(nsToPods[p.Namespace], p)
+	}
```

Comment:
I don't think we need to create a map with a copy of all pods in here again. We could simply store the state that we care about which is `hostNetwork`, `name`, `namespace` and `TerminationGracePeriodSeconds`.

Reply:
I think I should explain the motivation behind. Pod Disruption Budgets (PDB) are namespaced and applied to pods that satisfy a given selector. When you try to evict more pods than the PDB allows, you need to wait some time and retry.  
Given the following scenario:
```
NS1:
   pod1
   pod2
   pod3

PDB_1 -> selects pods from NS1  (maxUnavailable = 1, aka 1 disruption at a time)

NS2:
   pod1
   pod2
```

We have two namespaces and one has a PDB defined. When getting the list of pods you get something like:
```
pod.Items = [ pod1_ns1, pod2_ns1, pod3_ns1, pod1_ns2, pod2_ns2 ]
```
As you iterate over `pod.Items`, assuming pods on `ns1` take long time to terminate, you delete pod1 and immediately right after try to delete pod2, but you would get throttled by PDB_1 and need to wait some time X until you can keep with the loop. You'll get again throttled when trying to evict pod3. With this method, you haven even started to delete pods from ns2, which may or may not have a PDB. So, for the sake of making progress in this loop, and knowing how PDBs work, we should group pods by namespace (and maybe also by selector) so make evictions in the overall cluster faster. 

**Note:** I just realized I took the simplification of just grouping by namespace but in reality we should group by namespace and by selector. Otherwise, this will not work if you have all your cluster pods in a single namespace (e.g.: `default`) where you can actually have multiple PDBs that select different pods within the same namespace

Reply:
I'm fine with the motivation of grouping pods per namespace. My feedback is regarding the unnecessary memory consumption of duplicating a full pod structure while we can simply copy the fields that we care about.

Reply:
Oh ok, sorry for the confusion, I thought you were questioning the need for grouping in general. I think we can do that. Only that, if we copy those fields, we have to put them into a struct and I would prefer that we don't introduce throw-away structs just to keep things simple and easy to read. What about we take a pointer to those pods ? 

```
// current
var map[string][]corev1.Pod
// proposed
var map[string][]*corev1.Pod
```

wdyt ?

Reply:
yeah pointers work as fine

Reply:
I would like to understand why you changed your mind with pointers. I think the intent on `ceMap` is to create a `set`  , whereas here, we're trying to do a `groupBy` operation with a subset of the pods values. Two different purposes. Are we discussing a matter of style here ? I'm confused because we started with the argument that `memory` is a problem and we end up now with a code that is more verbose (need to add local struct) and more memory (copying values that already exist in memory).  

> while instead of struct{}{} we would have hostNetwork and TerminationGracePeriodSeconds

we would also need to add the pod's Name.

Reply:
I pushed two commits:

[ed13f5a](https://github.com/cilium/cilium-cli/pull/240/commits/ed13f5a370f51511a6675db1df95499aa777e5ec): using pointers
[7e6f72a](https://github.com/cilium/cilium-cli/pull/240/commits/7e6f72a1a6967b899ba1db8ba34ab57145f6aca2): using structs

Maybe it would be more evident to see the differences between the two approaches. Cheers !

Reply:
Sorry, I completely miss the point of having them grouped by namespace as I was reviewing the PR thinking that we can't perform a podList for the entire cluster. However that can be fixed in a follow up PR. Using pointers for now is fine.

Reply:
I just reverted the last commit that uses a struct and now we are back to using pointers. Cheers